selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/git/GitCompareTest.java

```diff
@@ -65,181 +94,152 @@
-    git.setCursorToLine(2, LEFT_COMPARE_ST);
-    git.typeTextIntoGitCompareEditor("//***che***codenvy***");
-    git.waitExpTextIntoCompareLeftEditor("//***che***codenvy***");
-    git.clickOnGitCompareCloseButton();
-    askDialog.confirmAndWaitClosed();
-    git.waitGitCompareFormIsClosed();
+    git.waitExpTextIntoCompareLeftEditor("public class NewClass");
+    git.waitTextNotPresentIntoCompareRightEditor("public class NewClass");
+    git.closeGitCompareForm();
     git.closeGroupGitCompareForm();
-    projectExplorer.openItemByPath(
-        PROJECT_NAME + "/src/main/java/com/codenvy/example/spring/A.java");
-    editor.waitActive();
-    editor.waitTextIntoEditor("//***che***codenvy***");
 
-    // check the 'git compare' after commit
+    // check the compare warning dialog after commit
     projectExplorer.waitAndSelectItem(PROJECT_NAME);
-    menu.runCommand(TestMenuCommandsConstants.Git.GIT, TestMenuCommandsConstants.Git.ADD_TO_INDEX);
-    git.waitGitStatusBarWithMess(TestGitConstants.GIT_ADD_TO_INDEX_SUCCESS);
-    menu.runCommand(TestMenuCommandsConstants.Git.GIT, TestMenuCommandsConstants.Git.COMMIT);
-    git.waitAndRunCommit("Update Java class");
+    menu.runCommand(GIT, COMMIT);
+    git.waitAndRunCommit("Change files");
     git.waitGitStatusBarWithMess(TestGitConstants.COMMIT_MESSAGE_SUCCESS);
-    loader.waitOnClosed();
-    projectExplorer.waitAndSelectItem(
-        PROJECT_NAME + "/src/main/java/com/codenvy/example/spring/GreetingController.java");
-    menu.runCommand(
-        TestMenuCommandsConstants.Git.GIT,
-        TestMenuCommandsConstants.Git.Compare.COMPARE_TOP,
-        TestMenuCommandsConstants.Git.Compare.COMPARE_LATEST_VER);
-    loader.waitOnClosed();
+    menu.runCommand(GIT, COMPARE_TOP, COMPARE_LATEST_VER);
     warningDialog.waitWaitWarnDialogWindowWithSpecifiedTextMess(
         "There are no changes in the selected item.");
     warningDialog.clickOkBtn();
+  }
 
-    // check the 'git compare' for another local branch
+  @Test(priority = 2)
+  public void checkCompareWithBranch() {
     projectExplorer.waitProjectExplorer();
-    menu.runCommand(
-        TestMenuCommandsConstants.Git.GIT,
-        TestMenuCommandsConstants.Git.Compare.COMPARE_TOP,
-        TestMenuCommandsConstants.Git.Compare.COMPARE_WITH_BRANCH);
+    projectExplorer.waitAndSelectItem(PROJECT_NAME);
+
+    // check the 'Close' button
+    menu.runCommand(GIT, COMPARE_TOP, COMPARE_WITH_BRANCH);
+    git.waitGitCompareBranchFormIsOpen();
+    gitCompare.clickOnCloseBranchCompareButton();
+
+    // check the 'git compare' for another local branch
+    menu.runCommand(GIT, COMPARE_TOP, COMPARE_WITH_BRANCH);
     git.waitGitCompareBranchFormIsOpen();
     git.selectBranchIntoGitCompareBranchForm("newbranch");
     git.clickOnCompareBranchFormButton();
-    loader.waitOnClosed();
+    git.waitGroupGitCompareIsOpen();
+    git.waitExpTextInGroupGitCompare(TEXT_GROUP);
+
+    // check the 'Next' and 'Previous' diff buttons
```

Comment:
Could you, please, divide the code which check 'Next' button, from those which check 'Previous' button?